### ISSUE TITLE
Going to dependent crates then back makes the crate look like it depends on itself

### ISSUE THREAD

**Issue by troplin** [rust-lang/crates.io#306]
1. Go to the page of a crate that has some dependencies.
2. Click on the link of one of the dependencies
3. On the page of the dependecy, click on "dependent crates".
4. Click on the link to the original crate again
5. Look at the dependency links:
   - The dependency that we were following is now gone
   - The crate now seems to depend on itself

**Comment by tbu-** [comment]
This also happens with the back button:

1) Go to a crate (e.g. [itertools](https://crates.io/crates/itertools)).
2) Click a dependency (e.g. permutohedron).
3) Click the back button in the browser.

Expected result: Crate page for itertools, listing permutohedron 0.1 as dependency.
Actual result: Crate page for itertools, listing itertools 0.1 as the only dependency.

Tested in Firefox 45.0.2 and Chromium 50.0.2661.75.

**Comment by carols10cents** [comment]
Just tried this and this still happens. Not sure how to fix it though.

**Comment by locks** [comment]
Back button works properly for me, but I still see the problem with OP's reproduction.

**Comment by Turbo87** [comment]
the issue here is as follows:

when we visit `/crates/itertools` it will download the dependencies from `/api/v1/crates/itertools/0.6.0/dependencies` which includes:

```json
{
  "crate_id":"either",
  "id":186109,
  "kind":"normal",
  "req":"^1.0",
  "target":null,
  "version_id":49566
}
```

when we click on the `either` dependency on the page it will request data from `/api/v1/crates/either/reverse_dependencies` which includes:

```json
{
  "crate_id":"itertools",
  "id":186109,
  "kind":"normal",
  "optional":false,
  "req":"^1.0",
  "target":null,
  "version_id":49566
}
```

when we click the back button Ember Data will reuse the dependencies that it already downloaded for the `itertools` crate, and since the `reverse_dependencies` call overwrote the `dependency:186109` record it will show `itertools` instead.

tl;dr it seems wrong that the API is returning a different `crate_id` content for the record depending on how it was queried (deps vs. reverse deps).
